org.eclipse.scanning.api/src/org/eclipse/scanning/api/database/CompositeBean.java

```diff
@@ -21,55 +26,86 @@
  *
  */
 public class CompositeBean {
+	
+	private static final Logger logger = LoggerFactory.getLogger(CompositeBean.class);
 
 	/**
 	 * IMPORTANT: Do not use generics here, the data
 	 * is almost always heterogeneous.
 	 */
-	private Map<Operation, Collection<Object>> data;
+	private Map<Operation, Collection<Bean>> data;
     
     public CompositeBean() {
     	data = new LinkedHashMap<>(); // Order matters in the collection of items
     }
     
-    public CompositeBean(Operation type, Object... beans) {
+    /**
+     * 
+     * @param type
+     * @param originalObjectsOrBeans may be some Bean objects or the actual database objects (which will be wrapped with Bean)
+     */
+    public CompositeBean(Operation type, Object... originalObjectsOrBeans) {
     	this();
     	assertType(type);
-     	data.put(type, new ArrayList<>(Arrays.asList(beans)));
+  
+    	List<Bean> wrapped = new ArrayList<Bean>(originalObjectsOrBeans.length);

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). (sonar.java.source not set. Assuming 7 or greater.) [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2293)